include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -0,0 +1,1897 @@
+    //!\}
+
+    //!\cond
+    // this signals to range-v3 that something is a container :|
+    using allocator_type    = void;
+    //!\endcond
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr dynamic_bitset()                                   noexcept = default; //!< Defaulted.
+    constexpr dynamic_bitset(dynamic_bitset const &)             noexcept = default; //!< Defaulted.
+    constexpr dynamic_bitset(dynamic_bitset &&)                  noexcept = default; //!< Defaulted.
+    constexpr dynamic_bitset & operator=(dynamic_bitset const &) noexcept = default; //!< Defaulted.
+    constexpr dynamic_bitset & operator=(dynamic_bitset &&)      noexcept = default; //!< Defaulted.
+    ~dynamic_bitset()                                            noexcept = default; //!< Defaulted.
+
+    /*!\brief Construct from an `uint64_t`.
+     * \param[in] value The `uint64_t` to construct/assign from.
+     * \throws std::invalid_argument if value has bits set outside of [0, 58).
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/range/container/dynamic_bitset_construct_uint64_t.cpp
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * Throws std::invalid_argument value has a set bit past the 58 one, i.e. only bits in [0,58) may be set.
+     */
+    constexpr dynamic_bitset(uint64_t const value)
+    {
+        if (detail::popcount(value >> 58) != 0)
+            throw std::invalid_argument{"The dynamic_bitset can be at most 58 long."};
+        data.bits |= value;
+        data.size |= value ? detail::most_significant_bit_set(value) + 1 : 0u;
+    }
+
+    /*!\brief Construct from two iterators.
+     * \tparam begin_it_type Must model std::ForwardIterator and `value_type` must be constructible from
+     *                       the reference type of begin_it_type.
+     * \tparam   end_it_type Must model std::Sentinel.
+     * \param[in] begin_it Begin of range to construct/assign from.
+     * \param[in] end_it End of range to construct/assign from.
+     *
+     * \details
+     *
+     * ### Complexity
+     *
+     * Linear in the distance between `begin_it` and `end_it`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    template <std::ForwardIterator begin_it_type, std::Sentinel<begin_it_type> end_it_type>
+    //!\cond
+        requires std::Constructible<value_type, reference_t<begin_it_type>>
+    //!\endcond
+    constexpr dynamic_bitset(begin_it_type begin_it, end_it_type end_it) noexcept:
+        dynamic_bitset{}
+    {
+        assign(begin_it, end_it);
+    }
+
+    /*!\brief Construct from a different range.
+     * \tparam other_range_t The type of range to be inserted; must satisfy std::ranges::InputRange and `value_type`
+     *                       must be constructible from reference_t<other_range_t>.
+     * \param[in]      range The sequence to construct/assign from.
```

Comment:
I mean, leave it as is right now but for the future please do not align different doxygen entities. That looks just awful.